ldm/dream/devices.py

```diff
@@ -13,9 +13,12 @@ def choose_torch_device() -> str:
 def choose_autocast_device(device):
     '''Returns an autocast compatible device from a torch device'''
     device_type = device.type # this returns 'mps' on M1
+    # autocast only for cuda, but GTX 16xx have issues with it
     if device_type == 'cuda':
-        return device_type,autocast
-    elif device_type == 'cpu':
-        return device_type,nullcontext
+        device_name = torch.cuda.get_device_name()
+        if 'GTX 16' in device_name:
```

Comment:
Nice fix. Though to prevent potential name clashes in the future, can you check against a more specific 'GeForce GTX 1660'? (As for 1650, don't remember seeing reports but it could easily be a separate check)